python/rmm/__init__.py

```diff
@@ -52,4 +52,9 @@
     "unregister_reinitialize_hook",
 ]
 
+# Note that it is important for this attribute to be specified in this module
+# (not in another module imported into this one) so that setuptools can parse
+# it statically without requiring the package to be importable at the time
+# metadata is processed.
 __version__ = "23.12.00"
+__git_commit__ = ""
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏ Will setuptools only grab dynamic versions from the top-level package, or can we have these living in their own file?

That is, could the `pyproject.toml` change be:

```
[tool.setuptools.dynamic]
version = {attr = "rmm._version.__version__"}
```

Rationale (if it is possible): minimising the number of lines that are potential candidates for rewriting with `sed` seems like a good idea.

Reply:
Will try it out and let you know. I agree that this would be nice.

Reply:
Yup this works. Great call.